x-pack/plugins/reporting/server/lib/screenshots/observable.ts

```diff
@@ -66,12 +75,13 @@ export function getScreenshots$(
               logger.error(err);
               return Rx.of({ ...defaultSetupResult, error: err }); // allow failover screenshot capture
             }),
-            takeUntil(exit$),
+            takeUntil(unexpectedExit$),
             screen.getScreenshots()
           )
         ),
         take(opts.urlsOrUrlLocatorTuples.length),
-        toArray()
+        toArray(),
+        mergeMap((results) => close().pipe(mapTo(results)))
```

Comment:
We shouldn't have to call `close` since `createPage()` adds an inner subscriber that kills the Chromium process:
https://github.com/elastic/kibana/blob/f93fcea9d8a31dc77cba2dfcd432e7a34b115681/x-pack/plugins/reporting/server/browsers/chromium/driver_factory/index.ts#L161-L164

Reply:
Right, from local testing, the moment when the `Compiling PDF using...` and the moment when the close function is called can occur as:

-> Compiling PDF
-> Closing browser

When I think, for memory pressure safety, we should coerce

-> Closing browser
-> Compiling PDF

I'll dig into this a little bit more, but this is one of the areas a relatively large memory spike usually occurs. I'll try to get some definitive data to support this though!

Reply:
We discussed this over a sync and came up with clear reasoning to use this. The inner observable that adds clean-up logic to the closing of the browser driver subscription should be treated as _fallback_ logic in case there is an error that prevents _explicitly_ closing the browser (using `mergeMap` in the observable chain and waiting on the async `close()` method)